multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/v2/ResourceLiveParameterResolver.java

```diff
@@ -29,81 +29,78 @@ public ResourceLiveParameterResolver(DeploymentDescriptor deploymentDescriptor)
+                                                                                 resourceParameter));
     }
 
-    private void updateMapFromModule(Resource resource, Reference ref, Module module, String resourceParameterValue,
-                                     Map.Entry<String, Object> resourceParameter) {
+    private void updateConfigParameterFromModule(Reference reference, Module module, String resourceParameterValue,
+                                                 Map.Entry<String, Object> resourceParameter) {
         Optional<ProvidedDependency> moduleProvidedDependencyOpt = module.getProvidedDependencies()
                                                                          .stream()
-                                                                         .filter(dp -> dp.getName()
-                                                                                         .equals(ref.getDependencyName()))
+                                                                         .filter(providedDependency -> providedDependency.getName()
+                                                                                                                         .equals(reference.getDependencyName()))
                                                                          .findAny();
 
         if (moduleProvidedDependencyOpt.isPresent()) {
             Object requiredDependencyValue = moduleProvidedDependencyOpt.get()
                                                                         .getProperties()
-                                                                        .get(ref.getKey());
+                                                                        .get(reference.getKey());
 
             if (requiredDependencyValue instanceof Map) {
                 resolveModuleProvidedDependency(castToMap(requiredDependencyValue), module);
                 resourceParameter.setValue(requiredDependencyValue);
-            } else if (isStringValueReplaceable(requiredDependencyValue)) {
-                replaceValueInResourceParameter(module, resourceParameterValue, resourceParameter, ref,
+            } else if (resourceParameterValue instanceof String && doesParameterContainDefaultPlaceholder(requiredDependencyValue)) {
+                replaceValueInResourceParameter(module, resourceParameterValue, resourceParameter, reference,
                                                 castToString(requiredDependencyValue));
             }
         }
     }
 
-    private boolean isStringValueReplaceable(Object object) {
-        return object instanceof String && idleToLiveParameterPairs.keySet()
-                                                                   .stream()
-                                                                   .anyMatch(defaultParamPlaceholder -> castToString(object).contains(defaultParamPlaceholder));
+    private boolean doesParameterContainDefaultPlaceholder(Object object) {
```

Comment:
I would switch the method signature to take a String here and cast it when calling